fennel.lua

```diff
@@ -2329,30 +2329,36 @@ local function repl(options)
 
         -- adds partial key matches in tbl to the match list
         local function addPartials(input, tbl, prefix)
-          for k in pairs(tbl) do
-            if #matches >= 40 then break -- cap completions at 40 to avoid overwhelming
-            elseif input == k:sub(0, #input) then
-              table.insert(matches, prefix .. k)
+            for k in pairs(tbl) do
+                if tbl == env or tbl == env.___replLocals___ then
+                    k = scope.unmanglings[k] or k
+                end
+                if #matches >= 40 then break -- cap completions at 40
```

Comment:
Yes, that occurred to me back when I hardcoded the limit too. I can add that in this PR if you'd like to save on updates but the hardcoded behavior is what was previously in completion so it would also make sense as a separate PR. I'm good either way on that front, I don't mind adding it to this one.

Another alternative would be to have the internal completer return a stateful iterator function that a consumer could call as many times as they'd like to form a table.

Reply:
Yeah, let's push this off to a separate PR. An iterator would be more flexible, but I don't think the extra complexity is called for.